src/OrchardCore.Modules/OrchardCore.Feeds/Controllers/FeedController.cs

```diff
@@ -37,8 +37,7 @@ public async Task<ActionResult> Index(string format)
             var bestFormatterMatch = _feedFormatProviders
                 .Select(provider => provider.Match(context))
                 .Where(match => match != null && match.FeedBuilder != null)
-                .OrderByDescending(match => match.Priority)
-                .FirstOrDefault();
+                .MaxBy(match => match.Priority);
```

Comment:
What if there's a null

Reply:
i dont understand

Reply:
The first quey can handle the first or default if there's a null, what about `MaxBy` I might need to have a look

Reply:
thats not what FirstOrDefault means. it doesnt "default if there's a null" it "returns the default if no items match". in this case since we dont have a `FirstOrDefault()` (ie no match), it will return the default is the enumerable is empty.


MaxBy has the same result https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.maxby?view=net-8.0

> If the source sequence is empty and TSource is a nullable type, this method returns null. If the source sequence is empty and TSource is a non-nullable struct, such as a primitive type, an [InvalidOperationException](https://learn.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-8.0) is thrown.

Reply:
I just deleted my above comment, ya default returns empty. In your case how do you handle the null for both iterated item and the `FeedBuilder`?

Reply:
@SimonCropp